src/main/java/bg/libapp/libraryapp/web/BookController.java

```diff
@@ -0,0 +1,92 @@
+package bg.libapp.libraryapp.web;
+
+import bg.libapp.libraryapp.exceptions.AuthorNotFoundException;
+import bg.libapp.libraryapp.exceptions.BookNotFoundException;
+import bg.libapp.libraryapp.exceptions.GenreNotFoundException;
+import bg.libapp.libraryapp.model.dto.book.BookAddDTO;
+import bg.libapp.libraryapp.model.dto.book.BookViewDTO;
+import bg.libapp.libraryapp.service.BookService;
+import jakarta.validation.Valid;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.Set;
+
+@RestController
+@RequestMapping("/api/books")
+public class BookController {
+    private final BookService bookService;
+
+    @Autowired
+    public BookController(BookService bookService) {
+        this.bookService = bookService;
+    }
+
+    @PostMapping("/add")
```

Comment:
No need for /add in the URL. HTTP Post on /books should be enough to know that you are creating a new book.